### PR TITLE
Lidar lite driver fixes.

### PR DESCRIPTION
**Pull request by jgoppert**
Cleans up pointer problems with lidar lite driver.

### CODE REVIEW

**Comment by jgoppert** [comment]
This is the build error, looks travis related:

```bash
@ clang++ -MD -c -fno-exceptions -fno-rtti -std=c++0x -fno-threadsafe-statics -D__CUSTOM_FILE_IO__ -g -Wall -Wextra -Werror -Wfloat-equal -Wpointer-arith -Wmissing-declarations -Wpacked -Wno-unused-parameter -Wno-packed -Werror=format-security -Werror=array-bounds -Wfatal-errors -Werror=unused-variable -Werror=reorder -Werror=uninitialized -Werror=init-self -Wno-gnu-array-member-paren-init -Wno-missing-field-initializers -Os -g3 -fno-strict-aliasing -fomit-frame-pointer -funsafe-math-optimizations -fno-builtin-printf -ffunction-sections -fdata-sections -DCONFIG_ARCH_BOARD_POSIXTEST -D__PX4_LINUX -D__PX4_POSIX -Dnoreturn_function= -I/home/travis/build/PX4/Firmware//src/modules/systemlib -I/home/travis/build/PX4/Firmware//src/lib/eigen -I/home/travis/build/PX4/Firmware//src/platforms/posix/include -I/home/travis/build/PX4/Firmware//mavlink/include/mavlink -Wno-error=shadow -DCONFIG_WCHAR_BUILTIN -Wno-effc++ -I/home/travis/build/PX4/Firmware/src/ -I/home/travis/build/PX4/Firmware/src//modules/ -I/home/travis/build/PX4/Firmware/src/include/ -I/home/travis/build/PX4/Firmware/src/lib/ -I/home/travis/build/PX4/Firmware/src/platforms/ -I/home/travis/build/PX4/Firmware/src/drivers/boards/posix -Wno-deprecated-register -Wno-tautological-constant-out-of-range-compare -Wno-unused-private-field -Wno-unused-const-variable -fvisibility=hidden -include /home/travis/build/PX4/Firmware/src/include/visibility.h /home/travis/build/PX4/Firmware/src/modules/systemlib/circuit_breaker.cpp -o circuit_breaker.cpp.o
CC:      /home/travis/build/PX4/Firmware/src/modules/systemlib/circuit_breaker_params.c
make[3]: *** No rule to make target `/home/travis/build/PX4/Firmware/Build/git_version.c', needed by `/home/travis/build/PX4/Firmware/Build/git_version.c.o'.  Stop.
make[3]: Leaving directory `/home/travis/build/PX4/Firmware/Build/posix_default.build/home/travis/build/PX4/Firmware/src/modules/systemlib'
make[2]: *** [/home/travis/build/PX4/Firmware/Build/posix_default.build//home/travis/build/PX4/Firmware/src/modules/systemlib/module.pre.o] Error 2
make[2]: Leaving directory `/home/travis/build/PX4/Firmware/Build/posix_default.build'
make[1]: *** [/home/travis/build/PX4/Firmware/Build/posix_default.build/firmware.a] Error 2
make[1]: Leaving directory `/home/travis/build/PX4/Firmware'
make: *** [posix] Error 2
```

File: ROMFS/px4fmu_common/init.d/rc.sensors [hunk 1]
```diff
@@ -115,7 +115,11 @@ if px4flow start
 then
 fi
 
-if ll40ls start
+if pwm_input start
```
Comment (mhkabir) [inline ROMFS/px4fmu_common/init.d/rc.sensors]: I think we should add a param like LL40LS_ENABLED here?

**Comment by jgoppert** [comment]
This is already a PR to master: https://github.com/PX4/Firmware/pull/2231

**Comment by TSC21** [comment]
Nice!

File: ROMFS/px4fmu_common/init.d/rc.sensors (hunk 1 above)
Comment (LorenzMeier) [inline ROMFS/px4fmu_common/init.d/rc.sensors]: Yes, it would be good to make the startup conditional.

**Comment by mhkabir** [comment]
@jgoppert Please merge conditional startup commits.
@LorenzMeier Conditional startup done.

**Comment by LorenzMeier** [comment]
Will merge once they are in.

**Comment by jgoppert** [comment]
Wait till I confirm the log is working now.

**Comment by jgoppert** [comment]
Lots of flight testing tonight, altitude hold is working great, no lidar problems. The log of the lidar distance was truncating to integers, I just changed it to floats instead of using cm, will test to make sure the log change works first thing in the morning.

**Comment by jgoppert** [comment]
Log fix confirmed.

lidar2

**Pull request merged**